app-android/src/main/java/io/github/droidkaigi/confsched2023/AppModule.kt

```diff
@@ -15,8 +18,25 @@ class AppModule {
     @Singleton
     @AppAndroidBuildConfig
     fun provideBuildConfigProvider(): BuildConfigProvider = AppBuildConfigProvider()
+
+    @Provides
+    @Singleton
+    fun provideNavigationRequester(): NavigationRequester = DefaultNavigationRequester()
 }
 
 class AppBuildConfigProvider(
     override val versionName: String = BuildConfig.VERSION_NAME,
 ) : BuildConfigProvider
+
+class DefaultNavigationRequester : NavigationRequester {
+    private val _navigateRequestStateFlow = MutableStateFlow("")
+    override fun getNavigationRouteFlow(): Flow<String> = _navigateRequestStateFlow
+
+    override fun navigateTo(route: String) {
+        _navigateRequestStateFlow.value = route
+    }
+
+    override fun navigated() {
+        _navigateRequestStateFlow.value = ""
+    }
+}
```

Comment:
We did this to operate between different NavHosts.
I don't think this is the best way to do it, so if you have a better idea, I'm open to it.